modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java

```diff
@@ -348,6 +376,16 @@ public PNGImageWriteParam() {
     private ROI roi;
 
     /**
+     * The NoData range to check nodata, or {@code null} if none.
+     */
+    private Range nodata;
+
+    /**
+     * Array of values used for indicating the background values
+     */
+    private double[] destNoData;
```

Comment:
The first one represents the raster Nodata while the array represents background values (each one per band). Actually JAI-EXT does not support a per band nodata, so only one value (inside the Range) can be set as NoData. Actually the nodata is set as the first value of the array (if the array is defined and input NoData are present).